packages/blade/src/components/Collapsible/_decisions/decisions.md

```diff
@@ -0,0 +1,186 @@
+
+### Accordion close
+
+https://github.com/razorpay/blade/assets/6682655/b54eaeb9-7257-4da4-bf3b-89e10785e4f9
+
+## API
+
+Annotated component with props ([figma](https://www.figma.com/file/cX625jYMAX1Roq8hLifFrn/Collapsible-breakdown?type=design&node-id=0-1&t=2fpp9pyvcuMxVTjW-0)):
+
+![Collapsible breakdown](./collapsible-breakdown.png)
+
+```jsx
+import { Collapsible, CollapsibleButton, CollapsibleLink, CollapsibleBody } from '@razorpay/blade';
+
+<Collapsible>
+  <CollapsibleButton>Button</CollapsibleButton>
+  <CollapsibleBody>
+    <Box>Slot</Box>
+  </CollapsibleBody>
+</Collapsible>;
+
+<Collapsible>
+  <CollapsibleLink>Button</CollapsibleLink>
+  <CollapsibleBody>
+    <Box>Slot</Box>
+  </CollapsibleBody>
+</Collapsible>;
+```
+
+> **Note**
+>
+> - `Collapsible` will compose (`CollapsibleButton` or `CollapsibleLink`), `CollapsibleBody`. This ensures all interactions, a11y and animations work seamlessly at user end.
+> - `CollapsibleButton` and `CollapsibleLink` will support a subset of relevant props from `Button` and `Link` respectively.
+
+### Collapsible
+
+| Prop              | Type            | Default     | Description                                                        | Required |
+| ----------------- | --------------- | ----------- | ------------------------------------------------------------------ | -------- |
+| children          | `JSX`           | `undefined` | Composes `CollapsibleButton`, `CollapsibleLink`, `CollapsibleBody` | ✅       |
+| direction         | `bottom`, `top` | `bottom`    | Direction in which the content expands                             |          |
+| defaultIsExpanded | `boolean`       | `false`     | Expands the collapsible content by default (uncontrolled)          |          |
+| isExpanded        | `boolean`       | `undefined` | Expands the collapsible content (controlled)                       |          |
+| onChange          | `function`      | `undefined` | Callback for change in collapsible's expanded state                |          |
+
+> **Note**
+>
+> - Also includes layout based styling props
+> - For using collapsible in controlled state, use `isExpanded` coupled with `onChange`
+> - `onChange` callback signature `({ isExpanded }) => {}`, `isExpanded` represents the collapsible's current state
+
+### CollapsibleButton
+
+Built-in trigger for `Collapsible`, provides a subset of `Button` [properties](../../Button/_decisions/decisions.md).
+
+Supports ✅
+
+- `children`
+- `variant`
+- `size`
+- `icon`
+- `iconPosition`
+- `isDisabled`
+- `testID`
+- `accessibilityLabel`
+
+Doesn't support ❌
+
+- `isFullWidth` - can visually break UI
+- `onClick`, `isLoading` - using `onClick` will break built-in interactions (unless we provide additional hooks), `onChange` should already provide a way for users to listen-in on `Collapsible` state. This might have usage for async with `isLoading` in the future but for now not exposing the prop
+- `type` - will be `button` always
```

Comment:
We anyway need a `CollapsibleLink` sub-component so it can automatically handle the toggling wiring logic. Might be good to restrict these props over here.
Basically `type` will not be available at all on `CollapsibleLink` because that would always be `button` so we can just avoid exposing the prop altogether.